saskatoon/harvest/api.py

```diff
@@ -221,3 +223,173 @@ def list(self, request, *args, **kwargs):
                                  'title': _("New Person")
                                  }
                          })
+
+
+class StatsView(LoginRequiredMixin, views.APIView):
+
+    template_name = "app/stats.html"
+    permission_classes = [IsCoreOrAdmin]
+    # ? seasons that will be shown in the dropdown list
+    seasons = list(map(lambda x: x[0], HarvestFilter.seasons)) + ["all"]
+
+    def get(self, request, format="html", *args, **kwargs):
+        """returns statistics on harvests for all seasons or a specific season"""
+
+        season = self.kwargs.get("season", "all")
```

Comment:
ok I think using a slug parameter passed as kwargs for `season` works. In terms of scalability (e.g. if one day we decide we want to also filter by city or other fields) I still think using a `GenericAPIView` (which can handle queryset / filters) could be suited. For example you could reuse the season filter defined in HarvestFilter and retrieve the filtered queryset with something like this:

```
self.harvest_queryset = self.filter_queryset(self.get_queryset()) 
self.harvest_yield_queryset = HarvestYield.filter(harvest__in=harvest_qs)

```
and work from these two querysets in your functions.
That being said it might not be as straightforward because you are not working with a single model (Harvest + HarvestYield) so my comment might not be accurate. I'll let you evaluate if you think it's worth doing it, otherwise absolutely no worries if it works as such!